src/Sulu/Bundle/AdminBundle/Resources/js/components/RectangleSelection/RectangleSelection.js

```diff
@@ -109,35 +176,102 @@ class RectangleSelection extends React.Component<Props> {
     };
 
     render() {
-        const {containerHeight, containerWidth, minHeight, minWidth, value = this.getMaximumSelection()} = this.props;
-        const {height, left, top, width} = value;
+        const {
+            backdrop,
+            containerHeight,
+            containerWidth,
+            disabled,
+            label,
+            minHeight,
+            minSizeNotification,
+            minWidth,
+            onFinish,
+        } = this.props;
+        const {height, left, top, width} = this.value;
 
         let backdropSize = 0;
-
-        if (containerHeight && containerWidth) {
+        if (backdrop && containerHeight && containerWidth) {
             backdropSize = Math.max(containerHeight, containerWidth);
         }
 
+        let minSizeReached = false;
+        if (minSizeNotification) {
+            if (height <= (minHeight || 0) && width <= (minWidth || 0)) {
+                minSizeReached = true;
+            }
+        }
+
+        return (
+            <ModifiableRectangle
+                backdropSize={backdropSize}
+                disabled={disabled}
+                height={height}
+                label={label}
+                left={left}
+                minSizeReached={minSizeReached}
+                onChange={this.handleRectangleChange}
+                onDoubleClick={this.handleRectangleDoubleClick}
+                onFinish={onFinish}
+                top={top}
+                width={width}
+            />
+        );
+    }
+}
+
+const RectangleSelectionWrapper = withPercentageValues(RectangleSelectionComponent);
+
+class RectangleSelectionContainer extends React.Component<Props> {
+    static defaultProps = {
+        backdrop: true,
+        disabled: false,
+        minHeight: undefined,
+        minSizeNotification: true,
+        minWidth: undefined,
+        round: true,
+        usePercentageValues: false,
+    };
+
+    render() {
+        const {children, ...rest} = this.props;
+
         return (
             <div className={rectangleSelectionStyles.selection}>
-                {this.props.children}
-                <ModifiableRectangle
-                    backdropSize={backdropSize}
-                    height={height}
-                    left={left}
-                    minSizeReached={height <= (minHeight || 0) && width <= (minWidth || 0)}
-                    onChange={this.handleRectangleChange}
-                    onDoubleClick={this.handleRectangleDoubleClick}
-                    top={top}
-                    width={width}
-                />
+                {children}
+                <RectangleSelectionWrapper {...rest} />
             </div>
         );
     }
 }
 
-export {
-    RectangleSelection,
-};
+// This export should only be used in tests
+export {RectangleSelectionContainer};
+
+const RectangleSelectionContainerWrapper = withContainerSize(
```

Comment:
`RectangleSelectionWithContainerSize`

Reply:
Continue discussion in this thread: https://github.com/sulu/sulu/pull/5434#discussion_r498299726